docker/Dockerfile.server

```diff
@@ -9,6 +9,19 @@ ARG EXTRA
 
 RUN python3 -m pip install --upgrade pip
 
+# Pre-install some requirements so that installing from the
+# wheel takes less time. This improves caching behavior
+# during development iterations, but should not be used
+# for production builds, as it will install extra dependencies
+# that aren't actually needed for the server to run.
+# The most convenient way to leverage this is by using a script
+# that automatically uncomments these lines and builds using the resulting
+# Dockerfile, but you can manually uncomment them.
+# Just make sure to have a requirements-dev-cache.txt file
+# in the docker dir before uncommenting.
+# <uncomment for pre-install-cache> COPY ./requirements*.txt /requirements/
+# <uncomment for pre-install-cache> RUN pip install -r /requirements/requirements-dev-cache.txt
+
```

Comment:
How about we have a `Dockerfile.server.dev` file that does this (no need for the script to uncomment), and `serve-dev` uses that one instead?

Reply:
It might just be annoying to remember we need to keep them in sync. This ensures they always are, at the cost of a few extra lines in `serve-dev`. I don't feel strongly about it though; happy to change if you do.